app/services/AppealService.scala

```diff
@@ -90,12 +89,12 @@ class AppealService @Inject()(penaltiesConnector: PenaltiesConnector,
     val enrolmentKey = constructMTDVATEnrolmentKey(userRequest.vrn)
     val appealType = userRequest.session.get(SessionKeys.appealType)
     val isLPP = appealType.contains(PenaltyTypeEnum.Late_Payment.toString) || appealType.contains(PenaltyTypeEnum.Additional.toString)
-    val agentDetails = getAgentDetails
+    val agentReferenceNo = userRequest.arn.getOrElse("")
     val penaltyId = userRequest.session.get(SessionKeys.penaltyId).get
     for {
       amountOfFileUploads <- uploadJourneyRepository.getNumberOfDocumentsForJourneyId(userRequest.session.get(SessionKeys.journeyId).get)
       modelFromRequest: AppealSubmission = AppealSubmission
-        .constructModelBasedOnReasonableExcuse(reasonableExcuse, isLateAppeal, amountOfFileUploads, agentDetails)
+        .constructModelBasedOnReasonableExcuse(reasonableExcuse, isLateAppeal, amountOfFileUploads, Some(agentReferenceNo))
```

Comment:
If you action the above, you can remove the Some(...)
```suggestion
        .constructModelBasedOnReasonableExcuse(reasonableExcuse, isLateAppeal, amountOfFileUploads, agentReferenceNo)
```